src/mjan.c

```diff
@@ -116,16 +120,45 @@ MOVELIST *GenerateMoves(POSITION position) {
   generic_hash_unhash(position, board);
 
   char piece = (generic_hash_turn(position) == 1) ? WCHAR : BCHAR;
-  int diffs[4] = {4, -4, 1, -1}; // up down right left
+  int diffs[8] = {4, -4, 1, -1, 5, -5, 3, -3};
+
+  // Main diagonal positions
+  BOOLEAN isMainDiagonalPos[16] = {
```

Comment:
what about instead of this we use something like
`#define MAINDIAGPOS(idx, jdx) (idx == jdx || idx + jdx == BOARDSIZE - 1)`

Reply:
I get what you're saying I think it'd have to be 
#define MAINDIAGPOS(idx, jdx) (idx == jdx || idx + jdx == (BOARDSIZE / 4) - 1)

Although, the current solution is less clever I feel like it's more readable so if it's okay with you I'm gonna keep the current.

Reply:
alr sg